apps/web/src/contexts/NftErrorContext.tsx

```diff
@@ -66,35 +74,37 @@ export function NftErrorProvider({ children }: PropsWithChildren) {
 
   const handleNftError = useCallback<NftErrorContextType['handleNftError']>(
     (tokenId: string, error: Error) => {
-      addBreadcrumb({ category: 'nft error', message: error.message, level: 'error' });
-
-      const token = { ...(tokens[tokenId] ?? defaultTokenErrorState()) };
-
-      token.isFailed = true;
-
-      // If the user refreshed the metadata and there was another failure,
-      // we'll show them a new toast telling them things failed to load,
-      // and we're looking into the issue asap.
-      if (token.refreshed) {
-        // Commenting this out because it's being displayed even without the user explicitly
-        // refreshing anything in the editor
-        //
-        // pushToast({
-        //   message:
-        //     'This piece has failed to load. This issue has been reported to the Gallery team.',
-        //   autoClose: true,
-        // });
-      }
-
       setTokens((previous) => {
         const next = { ...previous };
 
+        const token = { ...(next[tokenId] ?? defaultTokenErrorState()) };
+
+        if (error instanceof StillLoadingNftError) {
+          token.isLoading = true;
+        } else {
+          addBreadcrumb({ category: 'nft error', message: error.message, level: 'error' });
+          token.isFailed = true;
+          // If the user refreshed the metadata and there was another failure,
+          // we'll show them a new toast telling them things failed to load,
+          // and we're looking into the issue asap.
+          if (token.refreshed) {
+            // Commenting this out because it's being displayed even without the user explicitly
+            // refreshing anything in the editor
+            //
+            // pushToast({
+            //   message:
+            //     'This piece has failed to load. This issue has been reported to the Gallery team.',
+            //   autoClose: true,
+            // });
+          }
+        }
+
         next[tokenId] = token;
 
         return next;
       });
     },
-    [tokens]
+    []
```

Comment:
example here. rather than passing in `tokens`, better to use previous state passed into the state setter